examples/fastapi/tests.py

```diff
@@ -0,0 +1,35 @@
+from typing import Generator
+
+import pytest
+from fastapi.testclient import TestClient
+from main import app
+from models import Users
+
+from tortoise.contrib.test import initializer, finalizer
+
+
+@pytest.fixture(scope="module")
+def client() -> Generator:
+    initializer(
+        ["models",]
+    )
+    with TestClient(app) as c:
+        yield c
+    finalizer()
+
+
+def test_create_user(client: TestClient):
+    response = client.post("/users", json={"username": "admin"},)
+    assert response.status_code == 200, response.text
+    data = response.json()
+    assert data["username"] == "admin"
+    assert "id" in data
+    user_id = data["id"]
+
+    async def get_user_by_db():
+        user = await Users.get(id=user_id)
+        return user
+
+    event_loop = client.task.get_loop()
```

Comment:
Ah, I see. This is how we get the event loop.

Just wish we could have the top-level async, and the event loop not being messed up, so we don't have to do boilerplate heavy code like in here.

Reply:
Yes, maybe we can try pytest-asyncio ？

Reply:
I did try that a while ago, and it destroyed and recreated the event loop on every test, and that caused stale network connections to databases.
And I'm unsure how it will work with the starlette test client, as it's designed to be run with the event loop paused. So it may also just break.
My experience with py.test has been that it tries to do too much, so breaks asyncio a lot. e.g. pytest 5.4 has a regression that breaks asyncio contexts, which we depend upon.